service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt

```diff
@@ -369,6 +407,30 @@ private fun Database.Read.fetchChildPlacementBasics(childId: UUID, unitId: UUID)
         .list()
         .firstOrNull() ?: throw BadRequest("Child $childId has no placement in unit $unitId on the given day")
 }
+data class PlacementTypeDate(
+    val date: LocalDate,
+    val placementType: PlacementType
+)
+private fun Database.Read.fetchChildPlacementTypeDates(childId: UUID, unitId: UUID, startDate: LocalDate, endDate: LocalDate): List<PlacementTypeDate> {
+
+    // language=sql
+    val sql = """
+        SELECT d::date AS date, p.type AS placement_type
```

Comment:
I might add `SELECT DISTINCT` since the join to backup care may cause duplicates in rare cases.

Reply:
Shouldn't the `AND (p.unit_id = :unitId OR bc.unit_id = :unitId)` cover the backup care cases?

Reply:
I was thinking that in family units they do backup placements to the same unit as original placement, but maybe that still doesn't matter, not sure.